# 6. Delta Ladder

In [2]:
from fixedincome2025 import table

## Overview

* Recall that, given $D^{\$}$ and $C^{\$}$, we can estimate the change of the value of a bond portfolio for parallel yield curve shift
* As the yield curve doesn't always move parallelly, we need more granular control of the portfolio value for interest rate risk management
* The Delta ladder is a risk managing tool for this purpose

## Bump and Revaluate and Delta Ladder

* Having learned yield curve interpolation, we can obtain yield at any terms, and with that we can compute the value of arbitrary bonds/portfolios
* Today's curve is the base scenario, and the portfolio value computed with today's curve is the base PV (present value)
* Bumped curves:
    * There are 14 buckets on the below curve
    * Adding 1 bp to each bucket, we obtain 14 bumped curves
        * The first one has yields $\underline{4.24\%}, 4.18\%, 4.11\%, \ldots$
        * The second one has yields $4.23\%, \underline{4.19\%}, 4.11\%, \ldots$
* Revaluate the same portfolio with the 14 bumped curves to obtain 14 PVs
* These 14 PVs minus the base PV give you 14 PV differences, which form the Delta ladder

In [9]:
table('yc_10022025').T

1 Mo 1.5 Mo   2 Mo   3 Mo   4 Mo   6 Mo   1 Yr   2 Yr   3 Yr  \
10/2/2025  4.23%  4.18%  4.11%  4.02%  3.96%  3.81%  3.62%  3.55%  3.56%   

            5 Yr   7 Yr  10 Yr  20 Yr  30 Yr  
10/2/2025  3.67%  3.86%  4.10%  4.66%  4.69%

## Delta Ladder and PnL Computation

* Also known as the bucketed Delta, or index Delta

In [16]:
table('delta_ladder').T

1m  1.5m   2m   3m   4m   6m    1y    2y   3y  5y  7y  10y  20y  30y  \
Delta -185  -231 -303 -275 -690  778 -3739 -2320  314 -65  23   -6    1    0   

        Sum  
Delta -6698

* The risk in the above example is concentrated at at 1-2y part of the curve
* Profit and loss (PnL) computation:
    * Scenario 1: The 1m point of the curve is up 1 bp, while all other points unchanged
        * We will lose $\$185$ on the portfolio
    * Scenario 2: The 6m point is up 1 bp, while all other points unchanged
        * We will make a profit of $\$778$ on the portfolio
    * Scenario 3: The 1m and 6m points are both up 1 bp, while all other points unchanged
        * Portfolio PnL will be $778-185 = \$593$

## Delta Ladder and PnL Computation (Cont.)

In [16]:
table('delta_ladder').T

1m  1.5m   2m   3m   4m   6m    1y    2y   3y  5y  7y  10y  20y  30y  \
Delta -185  -231 -303 -275 -690  778 -3739 -2320  314 -65  23   -6    1    0   

        Sum  
Delta -6698

* PnL computation:
    * Scenario 4: The 1m point of the curve is up 2 bps, while all other points unchanged
        * We will lose $185\times 2 = \$370$ on the portfolio
    * Scenario 5: The 6m points is both up 1 bp and 1m point is up 2 bps, while all other points unchanged
        * Portfolio PnL will be $778-370 = \$408$
    * Scenario 6: Curve parallel shifts up by 1 bp
        * Portfolio PnL will be $-185-231-303 \cdots -6 + 1 + 0 = -6698$, sum of bucketed delta
        * For this reason, a bond portfolio's dollar duration $D^{\$}$ should be always close to its sum of bucketed delta

## Delta Ladder and PnL Computation: Exercise

In [3]:
table('yc_10022025_10032025')

10/2/2025 10/3/2025  Rates Movement (bps)  Portfolio Delta Ladder
1 Mo       4.23%     4.24%                     1                    -185
1.5 Mo     4.18%     4.17%                    -1                    -231
2 Mo       4.11%     4.11%                     0                    -303
3 Mo       4.02%     4.03%                     1                    -275
4 Mo       3.96%     3.96%                     0                    -690
6 Mo       3.81%     3.82%                     1                     778
1 Yr       3.62%     3.64%                     2                   -3739
2 Yr       3.55%     3.58%                     3                   -2320
3 Yr       3.56%     3.59%                     3                     314
5 Yr       3.67%     3.72%                     5                     -65
7 Yr       3.86%     3.90%                     4                      23
10 Yr      4.10%     4.13%                     3                      -6
20 Yr      4.66%     4.69%                     3                       1
30 Yr      4.69%     4.71%                     2                       0

## Delta Ladder and PnL Computation: Answer

* PnL is the rate movement times delta, bucket by bucket, and sum up the products
$$
(1)\times(-185) + (-1)\times(-231) + (0)\times(-303) + \cdots + (3)\times(1) + 2\times0 = -\$13,195
$$